src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -1074,7 +1074,7 @@ ClassPathEntry* find_first_module_cpe(ModuleEntry* mod_entry,
 
 
 // Search either the patch-module or exploded build entries for class.
-ClassFileStream* ClassLoader::search_module_entries(Thread* current,
+ClassFileStream* ClassLoader::search_module_entries(JavaThread* current,
```

Comment:
I wonder if these CDS and classLoader.hpp/cpp changes can be done first?  Just change the caller to pass ->as_Java_thread() until TRAPS is changed.